Added an es6 method for _.sortBy and _.orderBy

HermannBjorgvin opened the pull request:
This implementation is more limited than both Underscore and Lodash but worked in my case for what I had to do which was just to simply sort a list of objects alphabetically by an object key.

I don't have any information about browser compatibility, I guess the arrow functions and let keyword are the only es6 parts of this.

Cheers, thanks for the useful repo.

cht8687 commented on README.md:
```diff
@@ -117,7 +117,8 @@ then Lodash/Underscore is the better option.*
 1. [_.reduceRight](#_reduceright)
 1. [_.size](#_size)
-1. [_.some](#_some)
+1. [_.some](#_some))
```
There is an extra ')' after some

cht8687 replied:
_.orderBy seems missing from the quick links.

cht8687 commented on README.md:
```diff
@@ -117,7 +117,8 @@ then Lodash/Underscore is the better option.*
 1. [_.size](#_size)
-1. [_.some](#_some)
+1. [_.some](#_some))
+1. [_.sortBy](#_sortBy)
```
(#_sortBy) need to be changed to (#_sortby) to make the quick link work.

cht8687 commented on README.md:
```diff
@@ -1126,6 +1127,35 @@ Tests whether any of the elements in the array pass the test implemented by the
+  const fruits = [
+    {name:"banana", amount: 2},
+    {name:"apple", amount: 4},
+    {name:"pineapple", amount: 2}
```
missing ',' in the code.

cht8687 commented on README.md:
```diff
@@ -1126,6 +1127,35 @@ Tests whether any of the elements in the array pass the test implemented by the
+  // => [{name:"apple", amount: 4}, {name:"banana", amount: 2}, {name:"mango", amount: 1}, {name:"pineapple", amount: 2}]
+
+  // Native
+  let sortBy = (key) => {
```
maybe use `const` is better here since sortBy never changed.

HermannBjorgvin replied:
Yes definitely makes more sense.

cht8687 commented:
Hi @HermannBjorgvin , thanks for the contribution. Please check those comments which I made.

HermannBjorgvin commented:
Alright so I fixed the linking issues you mentioned. I changed let sortBy to const. Fixed those couple typos. I also changed the header to "_.sortBy and _.orderBy" but kept the two seperate links so it's easier to find alphabetically.

cht8687 commented on README.md:
```diff
@@ -116,8 +116,10 @@ then Lodash/Underscore is the better option.*
 1. [_.range](#_range)
 1. [_.reduce](#_reduce)
 1. [_.reduceRight](#_reduceright)
+1. [_.orderBy](#_sortby-and-_orderby)
```
A tiny last change - need move this _.orderBy to up, after _.minBy and before _.plunk ... ( as it's alphabetically listed)

HermannBjorgvin replied:
Ah shoot, moved it to the right spot.

HermannBjorgvin commented:
@cht8687 fixed that alphabetical ordering :+1:

petermikitsh commented:
It might be worth noting that lodash (haven't checked for underscore) returns a **copy** of the array (sorted), whereas the native sort API **mutates** the original.

If you want to mimic the behavior of lodash, you'd want to do (for example):

```js
const sortedArr = [...arr].sort(sortFn);
```

I used this guide when transitioning off of lodash, and that caveat would have been helpful to know, since I ended up introducing a regression into my code.

HermannBjorgvin commented:
Sorry about that @petermikitsh, still I'm honored that you're the first person I know of that I've introduced a regression into their code :)
